src/lib/jwt/index.ts

```diff
@@ -0,0 +1,131 @@
+import axios from "axios";
+import * as jose from "jose";
+import { NextRequest, NextResponse } from "next/server";
+
+import { UserToken } from "src/@types";
+
+import TokenExpired from "../exceptions/ExpiredToken";
+import InvalidToken from "../exceptions/InvalidToken";
+
+interface UserTokenPayload extends jose.JWTPayload {
+  data: string;
+}
+
+async function verifyAndUpdateToken(
+  req: NextRequest,
+  res: NextResponse,
+): Promise<{ response: NextResponse; isAuthenticated: Boolean }> {
+  const tokenPayload = (await req.cookies.get("jwtToken")?.value) || "";
+
+  if (!tokenPayload) {
+    res = NextResponse.redirect(new URL("/login", req.nextUrl));
```

Comment:
Sanitize user input:

```suggestion
  const validToken = /^[\w-]+\.[\w-]+\.[\w-]+$/.test(tokenPayload);
  if (!tokenPayload || !validToken) {`
    res = NextResponse.redirect(new URL("/login", req.nextUrl));
    res.cookies.delete("jwtToken");
```